### ISSUE TITLE
Cannot paginate within iframe

### ISSUE THREAD

**Issue by hubitor** [MontFerret/ferret#424]
Is it possible to paginate within an iframe?
I tried  several times with the CLICK() function as well as with the PAGINATION() function but neither of them worked. 
Example for CLICK():

```
LET page = DOCUMENT("https://www.w3schools.com/html/html_iframe.asp", {
    driver: "cdp",
    userAgent: "Mozilla/5.0 (Macintosh; Intel Mac OS X 10_14_6) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/76.0.3809.87 Safari/537.36"
})
WAIT(3000)
LET nextBtnSelector = 'a.w3-right.w3-btn'
LET frame1 = page.frames[1]
LET frameTitleSelector = '#main > h1 > span'
LET frameTitle = ELEMENT(frame1, frameTitleSelector).innerText
PRINT(frameTitle)
PRINT(ELEMENT(frame1, "#main > div.w3-panel.w3-info.intro > p:nth-child(1)").innerText)
CLICK(frame1, nextBtnSelector)
WAIT(3000)
//WAIT_ELEMENT(frame1, "#main > ul:nth-child(7) > li:nth-child(7)", 10000)
PRINT(frameTitle)
RETURN "DONE"
```

I see in the logs only the title of the first page ("Tutorial"). Ferret doesn't move to the next page ("Introduction").

**Comment by ziflex** [comment]
Hey, thanks for diving in!

Here is a query that is doing almost what you want:

```
LET page = DOCUMENT("https://www.w3schools.com/html/html_iframe.asp", { driver: "cdp" })

LET frames = (
    FOR frame IN page.frames
        FILTER frame.url == "https://www.w3schools.com/html/default.asp"
        LIMIT 1
        RETURN frame
)

LET frame = frames[0]

LET nextBtnSelector = 'a.w3-right.w3-btn'
LET frameTitleSelector = '#main > h1 > span'
LET frameTitle = ELEMENT(frame, frameTitleSelector).innerText

RETURN frameTitle
```

I guess the problem that you are accessing a particular frame by its position which might be wrong.

Regarding the click event, I think there is a bug in Ferret, since it does not update the iframe when it navigates. It will be fixed in the next release (0.10) soon.

**Comment by hubitor** [comment]
Thanks, but this is giving as result only the first page. I would like to extract data from all pages of the iframe.

I'll wait for the fix.
